internal/driver/controllerserver_helper.go

```diff
@@ -719,13 +665,17 @@ func (cs *ControllerServer) attachVolume(ctx context.Context, volumeID, linodeID
 		PersistAcrossBoots: &persist,
 	})
 	if err != nil {
-		code := codes.Internal // Default error code is Internal.
-		// Check if the error indicates that the volume is already attached.
+		// https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerpublishvolume-errors
 		var apiErr *linodego.Error
-		if errors.As(err, &apiErr) && strings.Contains(apiErr.Message, "is already attached") {
-			code = codes.Unavailable // Allow a retry if the volume is already attached: race condition can occur here
+		if errors.As(err, &apiErr) {
+			switch {
+			case strings.Contains(apiErr.Message, "is already attached"):
+				return errAlreadyAttached
+			case strings.Contains(apiErr.Message, "Maximum number of block storage volumes are attached to this Linode"):
```

Comment:
Ok, but this seems brittle when the API changes their text message. How can we check this error, but not rely on the spelling of an API text error message?

Reply:
That's an excellent question.
API doesn't document errors
![image](https://github.com/user-attachments/assets/1d392be3-8f9b-48bb-9fd1-3da96441fd48)
And linodego [report request as it](https://github.com/linode/linodego/blob/bb8fd62f542430bc465a75f3ba9a00293680db4b/volumes.go#L127C1-L131C2)

But we have to notice this logic was already done before in CSI :/
https://github.com/linode/linode-blockstorage-csi-driver/blob/17b7d1b66edf6762fabbc98c0ad944c417c309a7/internal/driver/controllerserver_helper.go#L725

Reply:
Yeah def not the ideal way to do it. Hopefully linodego can provide better error codes in future so we don't have to build logic based on the err strings :/